### PR TITLE
fix(developer): prevent error when scrolling touch layout editor with no selected key

### LINKED ISSUE #11107: bug(developer): scrolling touch layout editor can cause `TypeError: Cannot read property 'left' of undefined`
Sentry Issue: [KEYMAN-DEVELOPER-1J7](https://keyman.sentry.io/issues/4715313714/?referrer=github_integration)

```
TypeError: Cannot read property 'left' of undefined
  at Object.moveWedgesAround (/app/source/toucheditor/lib/builder.js:606:33)
  at HTMLDivElement.<anonymous> (/app/source/toucheditor/lib/builder.js:975:15)
  at HTMLDivElement.dispatch (/app/source/toucheditor/lib/jquery-1.10.2.js:5095:9)
  at HTMLDivElement.elemData.handle (/app/source/toucheditor/lib/jquery-1.10.2.js:4766:28)
```

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #11107.

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/tike/xml/layoutbuilder/builder.js [hunk 1]
```diff
@@ -18,8 +18,8 @@ $(function() {
   this.saveSelection = function() {
     let key = builder.selectedKey(), subKey = builder.selectedSubKey();
     return {
-      id: key ? $(key).data('id') : null,
-      subId: subKey ? $(subKey).data('id') : null
+      id: key.length ? $(key).data('id') : null,
```
Comment (darcywong00) [inline developer/src/tike/xml/layoutbuilder/builder.js:21]: Did this previously check if key or subKey are null?

Comment (mcdurdin) [reply developer/src/tike/xml/layoutbuilder/builder.js:21]: Yeah it turns out that `builder.selectedKey()` cannot return null -- the previous check was wrong. jquery returns an empty array (with its own extra props patched in) if there is no selected key.

jquery patterns are not ideal in this editor. But they are what we have

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.298-beta](https://keyman.com/downloads/releases/beta/17.0.298)

**Pull request merged**